### PR TITLE
chore: add schema validator

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link

None

### Change description

Add schema validator

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are commited to keeping commit history clean, consistent and linear. To achive this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +41,7 @@
     "@types/dotenv": "^8.2.0",
     "@types/express": "^4.0.37",
     "@types/express-session": "^1.15.3",
+    "@types/hapi__joi": "^16.0.9",
```
Comment (n-r-b) [inline package.json]: i don't think this one is being used?

File: package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "test": "mocha test/**/*.test.ts"
   },
   "dependencies": {
+    "@hapi/joi": "^17.1.0",
```
Comment (n-r-b) [inline package.json]: Should this be a developer dependency?

Comment (geek-soft) [reply package.json]: No, validation is needed at runtime. Anything that application needs to do the job should be defined as regular dependency.

File: package.json (hunk 1 above)
Comment (geek-soft) [reply package.json]: It is, the other one is not.

**Pull request merged**